autorel/clients/github_client.py

```diff
@@ -0,0 +1,146 @@
+import base64
+import re
+from clients.http_client import HTTPClient
+from clients.github_exceptions import (BadCredentialsException,
+                                       GithubException,
+                                       UserAgentException,
+                                       ObjectNotFoundException,
+                                       UnprocessableEntityException,
+                                       TwoFactorAuthException,
+                                       LoginAttemptExceededException,
+                                       BadRequestException,
+                                       RateLimitExceededException
+                                       )
+
+
+class Github:
```

Comment:
From Gitter:

--------------------
black-perl:
`Github` is going to have methods to access pull requests and commits. For eg: now I am doing this

```Python
>>> cli = Github(endpoint,port,login,password,None,None,user_agent,None)
>>> pull = PullRequests(cli)
```

So, I am passing cli to PullRequests. Now, I will add methods to the Github class and we can have something like this:

```Python
>>> cli = Github(endpoint,port,login,password,None,None,user_agent,None)
>>> pull = cli.pull_requests()
```
Similar, methods will be associated for Commits, Issues extra classes.

The method would be like this internally :
```Python
def pull_requests(self):
    pull_cli = PullRequests(self._client)
    return pull_cli
```

------------------

MrAnno:
Oh, okay! I like it this way.